<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>qSIPmg R package vignette</title>
  <meta name="description" content="This vignette explains the use of qSIPmg package for statisitical analyses and isotope incorporating features from a qSIP metagenomics study." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="qSIPmg R package vignette" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This vignette explains the use of qSIPmg package for statisitical analyses and isotope incorporating features from a qSIP metagenomics study." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="qSIPmg R package vignette" />
  
  <meta name="twitter:description" content="This vignette explains the use of qSIPmg package for statisitical analyses and isotope incorporating features from a qSIP metagenomics study." />
  

<meta name="author" content="Pranav Sampara, Kate Waring, and Ryan Ziels" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="background.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">qSIPmg vignette</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a></li>
<li class="chapter" data-level="2" data-path="background.html"><a href="background.html"><i class="fa fa-check"></i><b>2</b> Background</a><ul>
<li class="chapter" data-level="2.1" data-path="background.html"><a href="background.html#stable-isotope-probing"><i class="fa fa-check"></i><b>2.1</b> Stable isotope probing</a></li>
<li class="chapter" data-level="2.2" data-path="background.html"><a href="background.html#quantitative-metagenomics"><i class="fa fa-check"></i><b>2.2</b> Quantitative metagenomics</a></li>
<li class="chapter" data-level="2.3" data-path="background.html"><a href="background.html#quantitative-sip-qsip"><i class="fa fa-check"></i><b>2.3</b> Quantitative SIP (qSIP)</a></li>
<li class="chapter" data-level="2.4" data-path="background.html"><a href="background.html#for-details-on-sip-quantitative-metagenomics-and-qsip-please-refer-to-the-following-works"><i class="fa fa-check"></i><b>2.4</b> For details on SIP, quantitative metagenomics, and qSIP, please refer to the following works:</a><ul>
<li class="chapter" data-level="2.4.1" data-path="background.html"><a href="background.html#sip"><i class="fa fa-check"></i><b>2.4.1</b> SIP</a></li>
<li class="chapter" data-level="2.4.2" data-path="background.html"><a href="background.html#quantitative-metagenomics-1"><i class="fa fa-check"></i><b>2.4.2</b> Quantitative metagenomics</a></li>
<li class="chapter" data-level="2.4.3" data-path="background.html"><a href="background.html#qsip"><i class="fa fa-check"></i><b>2.4.3</b> qSIP</a></li>
<li class="chapter" data-level="2.4.4" data-path="background.html"><a href="background.html#htssip-r-package"><i class="fa fa-check"></i><b>2.4.4</b> HTSSIP R package</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html"><i class="fa fa-check"></i><b>3</b> Using qSIPmg R package</a><ul>
<li class="chapter" data-level="3.1" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#example-workflow"><i class="fa fa-check"></i><b>3.1</b> Example workflow</a></li>
<li class="chapter" data-level="3.2" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#load-required-data"><i class="fa fa-check"></i><b>3.2</b> Load required data</a><ul>
<li class="chapter" data-level="3.2.1" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#mag-coverage-data"><i class="fa fa-check"></i><b>3.2.1</b> MAG coverage data</a></li>
<li class="chapter" data-level="3.2.2" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#sequin-metadata"><i class="fa fa-check"></i><b>3.2.2</b> Sequin metadata</a></li>
<li class="chapter" data-level="3.2.3" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#dilutions-data"><i class="fa fa-check"></i><b>3.2.3</b> Dilutions data</a></li>
<li class="chapter" data-level="3.2.4" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#fractions-metadata"><i class="fa fa-check"></i><b>3.2.4</b> Fractions metadata</a></li>
<li class="chapter" data-level="3.2.5" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#gtdb-style-taxonomy-data"><i class="fa fa-check"></i><b>3.2.5</b> GTDB style taxonomy data</a></li>
<li class="chapter" data-level="3.2.6" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#mag-absolute-concentrations"><i class="fa fa-check"></i><b>3.2.6</b> MAG absolute concentrations</a></li>
<li class="chapter" data-level="3.2.7" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#gc-content"><i class="fa fa-check"></i><b>3.2.7</b> GC content</a></li>
<li class="chapter" data-level="3.2.8" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#log-scale-boolean"><i class="fa fa-check"></i><b>3.2.8</b> Log scale BOOLEAN</a></li>
<li class="chapter" data-level="3.2.9" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#coefficient-of-variation"><i class="fa fa-check"></i><b>3.2.9</b> Coefficient of variation</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#coverage-normalization-and-scaling"><i class="fa fa-check"></i><b>3.3</b> Coverage normalization and scaling</a><ul>
<li class="chapter" data-level="3.3.1" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#this-function-uses-the-following-datasets"><i class="fa fa-check"></i><b>3.3.1</b> This function uses the following datasets:</a></li>
</ul></li>
<li class="chapter" data-level="3.4" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#preparation-of-scaled-data-for-qsip-analysis"><i class="fa fa-check"></i><b>3.4</b> Preparation of scaled data for qSIP analysis</a></li>
<li class="chapter" data-level="3.5" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#estimation-of-atom-fraction-excess"><i class="fa fa-check"></i><b>3.5</b> Estimation of atom fraction excess</a></li>
<li class="chapter" data-level="3.6" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#plot-the-atom-fraction-excess-plot"><i class="fa fa-check"></i><b>3.6</b> Plot the atom fraction excess plot</a></li>
<li class="chapter" data-level="3.7" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#check-incorporator-list"><i class="fa fa-check"></i><b>3.7</b> Check incorporator list</a></li>
<li class="chapter" data-level="3.8" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#workflow-to-view-abundance-vs-bd-plots-of-incorporators"><i class="fa fa-check"></i><b>3.8</b> Workflow to view abundance vs BD plots of incorporators</a></li>
<li class="chapter" data-level="3.9" data-path="using-qsipmg-r-package.html"><a href="using-qsipmg-r-package.html#session-information"><i class="fa fa-check"></i><b>3.9</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html"><i class="fa fa-check"></i><b>4</b> Scaling coverage data</a><ul>
<li class="chapter" data-level="4.1" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#basic-approach-for-scaling"><i class="fa fa-check"></i><b>4.1</b> Basic approach for scaling</a></li>
<li class="chapter" data-level="4.2" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#choice-of-regression-model"><i class="fa fa-check"></i><b>4.2</b> Choice of regression model</a></li>
<li class="chapter" data-level="4.3" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#scaling-the-data"><i class="fa fa-check"></i><b>4.3</b> Scaling the data</a></li>
<li class="chapter" data-level="4.4" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#comparison-of-the-fits"><i class="fa fa-check"></i><b>4.4</b> Comparison of the fits</a></li>
<li class="chapter" data-level="4.5" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#decision-on-the-method-to-scale-coverages"><i class="fa fa-check"></i><b>4.5</b> Decision on the method to scale coverages</a></li>
<li class="chapter" data-level="4.6" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#session-information-1"><i class="fa fa-check"></i><b>4.6</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><i class="fa fa-check"></i><b>5</b> qSIP metagenomics and identification of isotope incorporators</a><ul>
<li class="chapter" data-level="5.1" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#why-perform-multiple-hypothesis-testing-with-fdr-correction"><i class="fa fa-check"></i><b>5.1</b> Why perform multiple hypothesis testing with FDR correction?</a></li>
<li class="chapter" data-level="5.2" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#comparison-of-the-two-methods"><i class="fa fa-check"></i><b>5.2</b> Comparison of the two methods</a><ul>
<li class="chapter" data-level="5.2.1" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#afe-method"><i class="fa fa-check"></i><b>5.2.1</b> AFE method</a></li>
<li class="chapter" data-level="5.2.2" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#multiple-testing"><i class="fa fa-check"></i><b>5.2.2</b> Multiple testing</a></li>
<li class="chapter" data-level="5.2.3" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#comparison-of-the-two-methods-1"><i class="fa fa-check"></i><b>5.2.3</b> Comparison of the two methods</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#session-information-2"><i class="fa fa-check"></i><b>5.3</b> Session information</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">qSIPmg R package vignette</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">qSIPmg R package vignette</h1>
<p class="author"><em>Pranav Sampara, Kate Waring, and Ryan Ziels</em></p>
<p class="date"><em>27/10/2021</em></p>
</div>
<div id="introduction" class="section level1">
<h1><span class="header-section-number">Chapter 1</span> Introduction</h1>
<p>The qSIPmg R package was designed to enable the exploration and statistical analysis of qSIP metagenomics data. Particularly, this package allows the identification of isotope incorporators and quantifies isotopic enrichment.</p>
<p>Please check out this introductory vignette to qSIP metagenomics, before checking out vignettes on:</p>
<p><a href="scaling-coverage-data.html#scaling-coverage-data">Coverage normalization and scaling based on linear regression model</a></p>
<p><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#qsip-metagenomics-and-identification-of-isotope-incorporators">Identification of incorporators based on atom fraction excess method or by multiple hypothesis testing on centroid buoyant densities</a></p>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="background.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
},
"link": "GitHub repo link"
});
});
</script>

</body>

</html>
