<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 qSIP metagenomics and identification of isotope incorporators | qSIPmg R package vignette</title>
  <meta name="description" content="This vignette explains the use of qSIPmg package for statisitical analyses and isotope incorporating features from a qSIP metagenomics study." />
  <meta name="generator" content="bookdown 0.24 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 qSIP metagenomics and identification of isotope incorporators | qSIPmg R package vignette" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="This vignette explains the use of qSIPmg package for statisitical analyses and isotope incorporating features from a qSIP metagenomics study." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 qSIP metagenomics and identification of isotope incorporators | qSIPmg R package vignette" />
  
  <meta name="twitter:description" content="This vignette explains the use of qSIPmg package for statisitical analyses and isotope incorporating features from a qSIP metagenomics study." />
  

<meta name="author" content="Pranav Sampara, Kate Waring, and Ryan Ziels" />



  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="scaling-coverage-data.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<script src="libs/accessible-code-block-0.0.1/empty-anchor.js"></script>
<link href="libs/anchor-sections-1.0.1/anchor-sections.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.0.1/anchor-sections.js"></script>
<link href="libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="libs/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #c4a000; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #000000; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #000000; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #000000; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */
</style>


<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">qSIPmg vignette</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#stable-isotope-probing"><i class="fa fa-check"></i><b>1.1</b> Stable isotope probing</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#quantitative-metagenomics"><i class="fa fa-check"></i><b>1.2</b> Quantitative metagenomics</a></li>
<li class="chapter" data-level="1.3" data-path="index.html"><a href="index.html#quantitative-sip-qsip"><i class="fa fa-check"></i><b>1.3</b> Quantitative SIP (qSIP)</a></li>
<li class="chapter" data-level="1.4" data-path="index.html"><a href="index.html#for-details-on-sip-quantitative-metagenomics-and-qsip-please-refer-to-the-following-works"><i class="fa fa-check"></i><b>1.4</b> For details on SIP, quantitative metagenomics, and qSIP, please refer to the following works:</a><ul>
<li class="chapter" data-level="1.4.1" data-path="index.html"><a href="index.html#sip"><i class="fa fa-check"></i><b>1.4.1</b> SIP</a></li>
<li class="chapter" data-level="1.4.2" data-path="index.html"><a href="index.html#quantitative-metagenomics-1"><i class="fa fa-check"></i><b>1.4.2</b> Quantitative metagenomics</a></li>
<li class="chapter" data-level="1.4.3" data-path="index.html"><a href="index.html#qsip"><i class="fa fa-check"></i><b>1.4.3</b> qSIP</a></li>
<li class="chapter" data-level="1.4.4" data-path="index.html"><a href="index.html#htssip-r-package"><i class="fa fa-check"></i><b>1.4.4</b> HTSSIP R package</a></li>
</ul></li>
<li class="chapter" data-level="1.5" data-path="index.html"><a href="index.html#example-workflow"><i class="fa fa-check"></i><b>1.5</b> Example workflow</a></li>
<li class="chapter" data-level="1.6" data-path="index.html"><a href="index.html#load-required-data"><i class="fa fa-check"></i><b>1.6</b> Load required data</a><ul>
<li class="chapter" data-level="1.6.1" data-path="index.html"><a href="index.html#mag-coverage-data"><i class="fa fa-check"></i><b>1.6.1</b> MAG coverage data</a></li>
<li class="chapter" data-level="1.6.2" data-path="index.html"><a href="index.html#sequin-metadata"><i class="fa fa-check"></i><b>1.6.2</b> Sequin metadata</a></li>
<li class="chapter" data-level="1.6.3" data-path="index.html"><a href="index.html#dilutions-data"><i class="fa fa-check"></i><b>1.6.3</b> Dilutions data</a></li>
<li class="chapter" data-level="1.6.4" data-path="index.html"><a href="index.html#fractions-metadata"><i class="fa fa-check"></i><b>1.6.4</b> Fractions metadata</a></li>
<li class="chapter" data-level="1.6.5" data-path="index.html"><a href="index.html#gtdb-style-taxonomy-data"><i class="fa fa-check"></i><b>1.6.5</b> GTDB style taxonomy data</a></li>
<li class="chapter" data-level="1.6.6" data-path="index.html"><a href="index.html#mag-absolute-concentrations"><i class="fa fa-check"></i><b>1.6.6</b> MAG absolute concentrations</a></li>
<li class="chapter" data-level="1.6.7" data-path="index.html"><a href="index.html#gc-content"><i class="fa fa-check"></i><b>1.6.7</b> GC content</a></li>
<li class="chapter" data-level="1.6.8" data-path="index.html"><a href="index.html#log-scale-boolean"><i class="fa fa-check"></i><b>1.6.8</b> Log scale BOOLEAN: </a></li>
<li class="chapter" data-level="1.6.9" data-path="index.html"><a href="index.html#coeefficient-of-variation"><i class="fa fa-check"></i><b>1.6.9</b> Coeefficient of variation: </a></li>
</ul></li>
<li class="chapter" data-level="1.7" data-path="index.html"><a href="index.html#coverage-normalization-and-scaling"><i class="fa fa-check"></i><b>1.7</b> Coverage normalization and scaling</a><ul>
<li class="chapter" data-level="1.7.1" data-path="index.html"><a href="index.html#this-function-uses-the-following-datasets"><i class="fa fa-check"></i><b>1.7.1</b> This function uses the following datasets:</a></li>
</ul></li>
<li class="chapter" data-level="1.8" data-path="index.html"><a href="index.html#preparation-of-scaled-data-for-qsip-analysis"><i class="fa fa-check"></i><b>1.8</b> Preparation of scaled data for qSIP analysis</a></li>
<li class="chapter" data-level="1.9" data-path="index.html"><a href="index.html#estimation-of-atom-fraction-excess"><i class="fa fa-check"></i><b>1.9</b> Estimation of atom fraction excess</a></li>
<li class="chapter" data-level="1.10" data-path="index.html"><a href="index.html#plot-the-atom-fraction-excess-plot"><i class="fa fa-check"></i><b>1.10</b> Plot the atom fraction excess plot</a></li>
<li class="chapter" data-level="1.11" data-path="index.html"><a href="index.html#check-incorporator-list"><i class="fa fa-check"></i><b>1.11</b> Check incorporator list</a></li>
<li class="chapter" data-level="1.12" data-path="index.html"><a href="index.html#workflow-to-view-abundance-vs-bd-plots-of-incorporators"><i class="fa fa-check"></i><b>1.12</b> Workflow to view abundance vs BD plots of incorporators</a></li>
<li class="chapter" data-level="1.13" data-path="index.html"><a href="index.html#session-information"><i class="fa fa-check"></i><b>1.13</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html"><i class="fa fa-check"></i><b>2</b> Scaling coverage data</a><ul>
<li class="chapter" data-level="2.1" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#basic-approach-for-scaling"><i class="fa fa-check"></i><b>2.1</b> Basic approach for scaling</a></li>
<li class="chapter" data-level="2.2" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#choice-of-regression-model"><i class="fa fa-check"></i><b>2.2</b> Choice of regression model</a></li>
<li class="chapter" data-level="2.3" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#scaling-the-data"><i class="fa fa-check"></i><b>2.3</b> Scaling the data</a></li>
<li class="chapter" data-level="2.4" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#comparison-of-the-fits"><i class="fa fa-check"></i><b>2.4</b> Comparison of the fits</a></li>
<li class="chapter" data-level="2.5" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#confirming-the-filtering-of-outliers-and-decrease-of-their-influence"><i class="fa fa-check"></i><b>2.5</b> Confirming the filtering of outliers and decrease of their influence</a></li>
<li class="chapter" data-level="2.6" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#decision-on-the-method-to-scale-coverages"><i class="fa fa-check"></i><b>2.6</b> Decision on the method to scale coverages</a></li>
<li class="chapter" data-level="2.7" data-path="scaling-coverage-data.html"><a href="scaling-coverage-data.html#session-information-1"><i class="fa fa-check"></i><b>2.7</b> Session information</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><i class="fa fa-check"></i><b>3</b> qSIP metagenomics and identification of isotope incorporators</a><ul>
<li class="chapter" data-level="3.1" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#why-perform-multiple-hypothesis-testing-with-fdr-correction"><i class="fa fa-check"></i><b>3.1</b> Why perform multiple hypothesis testing with FDR correction?</a></li>
<li class="chapter" data-level="3.2" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#comparison-of-the-two-methods"><i class="fa fa-check"></i><b>3.2</b> Comparison of the two methods</a><ul>
<li class="chapter" data-level="3.2.1" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#afe-method"><i class="fa fa-check"></i><b>3.2.1</b> AFE method</a></li>
<li class="chapter" data-level="3.2.2" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#multiple-testing"><i class="fa fa-check"></i><b>3.2.2</b> Multiple testing</a></li>
<li class="chapter" data-level="3.2.3" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#comparison-of-the-two-methods-1"><i class="fa fa-check"></i><b>3.2.3</b> Comparison of the two methods</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="qsip-metagenomics-and-identification-of-isotope-incorporators.html"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#session-information-2"><i class="fa fa-check"></i><b>3.3</b> Session information</a></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">qSIPmg R package vignette</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="qsip-metagenomics-and-identification-of-isotope-incorporators" class="section level1">
<h1><span class="header-section-number">Chapter 3</span> qSIP metagenomics and identification of isotope incorporators</h1>
<p>The isotope incorporators could be identified using two methods:</p>
<ol style="list-style-type: decimal">
<li>Atom fraction excess (AFE) method. E.g. the Hungate et al. (2015) AFE estimation method.</li>
<li>Multiple hypothesis testing to assess statistical significance</li>
</ol>
<p>While the first method gives a quantitative estimate of isotopic enrichment in a taxon to determine if it is an incorporator, the second method relies on null hypothesis testing of statistical significance between unlabeled and labeled treatments for a taxon.</p>
<p>In this vignette, we will compare the two methods using the data from the introductory vignette to qSIPmg.</p>
<p>Please refer this study, for the AFE method proposed by <a href="https://journals.asm.org/doi/10.1128/AEM.02280-15">Hungate et al. (2015)</a></p>
<p>For significance testing, we will use multiple hypothesis testing with Benjamini-Hohberg method to control the false discovery rate (FDR). FDR correction is performed based on the R package <a href="https://cran.r-project.org/web/packages/fdrtool/fdrtool.pdf">fdrtool</a>.</p>
<div id="why-perform-multiple-hypothesis-testing-with-fdr-correction" class="section level2">
<h2><span class="header-section-number">3.1</span> Why perform multiple hypothesis testing with FDR correction?</h2>
<p>For a detailed understanding of the issues with multiple testing, please refer the following works and more</p>
<ol style="list-style-type: decimal">
<li><p><a href="https://www.stat.berkeley.edu/~mgoldman/Section0402.pdf" class="uri">https://www.stat.berkeley.edu/~mgoldman/Section0402.pdf</a></p></li>
<li><p><a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1716-1" class="uri">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1716-1</a></p></li>
<li><p><a href="https://mb3is.megx.net/gustame/warnings/warning-multiple-testing" class="uri">https://mb3is.megx.net/gustame/warnings/warning-multiple-testing</a></p></li>
</ol>
<p>Briefly, when the same statistical test is used to test several hypotheses simultaneously, the chance of falsely rejecting the null hypothesis (false positive rate or the rate of type 1 error) increases. This can be controlled using several methods as discussed in the above links. Of these methods, FDR correction is less conservative in controlling the probability of false positives, and thus has greater statistical power than other familywise error rate control methods.</p>
<p>Ok, now let’s compare the two methods in identifying isotope incorporators</p>
</div>
<div id="comparison-of-the-two-methods" class="section level2">
<h2><span class="header-section-number">3.2</span> Comparison of the two methods</h2>
<div id="afe-method" class="section level3">
<h3><span class="header-section-number">3.2.1</span> AFE method</h3>
<p>Isotope incorporators from the introductory vignette were identified using the AFE method
Reproducing the code here, we have</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb1-1"></a>atomX =<span class="st"> </span><span class="kw">qSIP_atom_excess_MAGs</span>(phylo.qSIP,</span>
<span id="cb1-2"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb1-2"></a>                               <span class="dt">control_expr=</span><span class="st">&#39;Isotope==&quot;12C&quot;&#39;</span>,</span>
<span id="cb1-3"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb1-3"></a>                               <span class="dt">treatment_rep=</span><span class="st">&#39;Replicate&#39;</span>,</span>
<span id="cb1-4"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb1-4"></a>                               <span class="dt">Gi =</span> GC_content)</span>
<span id="cb1-5"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb1-5"></a><span class="co">#Bootstrap confidence intervals</span></span>
<span id="cb1-6"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb1-6"></a>df_atomX_boot =<span class="st"> </span><span class="kw">qSIP_bootstrap</span>(atomX, <span class="dt">n_boot=</span><span class="dv">100</span>, <span class="dt">parallel =</span> <span class="ot">TRUE</span>) <span class="co">#Change &quot;parallel = FALSE&quot; to compute using a single-core</span></span>
<span id="cb1-7"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb1-7"></a>CI_threshold =<span class="st"> </span><span class="dv">0</span></span>
<span id="cb1-8"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb1-8"></a>df_atomX_boot =<span class="st"> </span>df_atomX_boot <span class="op">%&gt;%</span></span>
<span id="cb1-9"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb1-9"></a><span class="st">  </span><span class="kw">mutate</span>(<span class="dt">Incorporator =</span> A_CI_low <span class="op">&gt;</span><span class="st"> </span>CI_threshold,</span>
<span id="cb1-10"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb1-10"></a>         <span class="dt">OTU =</span> <span class="kw">reorder</span>(OTU, <span class="op">-</span>A))</span></code></pre></div>
<p>List out the incorporators</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb2-1"></a><span class="co">#Get incorporator info</span></span>
<span id="cb2-2"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb2-2"></a>n_incorp_AFE =<span class="st"> </span>df_atomX_boot <span class="op">%&gt;%</span></span>
<span id="cb2-3"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb2-3"></a><span class="st">  </span><span class="kw">filter</span>(Incorporator <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>) <span class="op">%&gt;%</span></span>
<span id="cb2-4"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb2-4"></a><span class="st">  </span>nrow </span>
<span id="cb2-5"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb2-5"></a><span class="co">#Get incorporator list</span></span>
<span id="cb2-6"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb2-6"></a>incorporator_list_AFE =<span class="st"> </span><span class="kw">incorporators_taxonomy</span>(<span class="dt">taxonomy =</span> taxonomy_tibble, <span class="dt">bootstrapped_AFE_table =</span> df_atomX_boot)</span>
<span id="cb2-7"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb2-7"></a><span class="co">#Print incorporator information</span></span>
<span id="cb2-8"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb2-8"></a><span class="kw">cat</span>(<span class="st">&#39;Number of incorporators:&#39;</span>, n_incorp_AFE, <span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>)</span></code></pre></div>
<pre><code>## Number of incorporators: 25</code></pre>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb4-1"></a>rmarkdown<span class="op">::</span><span class="kw">paged_table</span>(incorporator_list_AFE)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["OTU"],"name":[1],"type":["fct"],"align":["left"]},{"label":["taxonomy"],"name":[2],"type":["chr"],"align":["left"]}],"data":[{"1":"NC_000913.3","2":"Terriglobusroseus"},{"1":"NC_003450.3","2":"Segniliparusrotundus"},{"1":"NC_008261.1","2":"Nocardiopsisdassonvillei"},{"1":"NC_010067.1","2":"Olsenellauli"},{"1":"NC_014008.1","2":"Meiothermus_Bsilvanus"},{"1":"NC_014168.1","2":"Meiothermus_Bsilvanus"},{"1":"NC_014210.1","2":"Hungateiclostridiumthermocellum"},{"1":"NC_014211.1","2":"Clostridium_Pperfringens"},{"1":"NC_014212.1","2":"Desulfosporosinusacidiphilus"},{"1":"NC_014213.1","2":"Desulfosporosinusacidiphilus"},{"1":"NC_014214.1","2":"Desulfosporosinusacidiphilus"},{"1":"NC_014363.1","2":"Desulfosporosinusmeridiei"},{"1":"NC_015761.1","2":"Streptococcuspyogenes"},{"1":"NC_017033.1","2":"Thermobacilluscomposti"},{"1":"NC_018014.1","2":"Hirschiabaltica"},{"1":"NC_019792.1","2":"Pseudomonas_Astutzeri_AE"},{"1":"NC_019897.1","2":"Pseudomonas_Astutzeri_AE"},{"1":"NC_019936.1","2":"Frateuriaaurantia"},{"1":"NC_019937.1","2":"Sediminispirochaetasmaragdinae"},{"1":"NC_019938.1","2":"Fervidobacteriumpennivorans"},{"1":"NC_019939.1","2":"Coraliomargaritaakajimensis"},{"1":"NC_019964.1","2":"Halovivaxruber"},{"1":"NC_019974.1","2":"Natronobacteriumgregoryi"},{"1":"NC_019975.1","2":"Natronococcusoccultus"},{"1":"NC_019976.1","2":"Natronococcusoccultus"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="multiple-testing" class="section level3">
<h3><span class="header-section-number">3.2.2</span> Multiple testing</h3>
<p>The atom fraction excess table obtained in the introductory vignette has the centroid buoyant densities as well. In this method, we will perform t-tests on the centroid buoyant densities. Centroid buoyant densities are weighted mean buoyant density for a taxon, the weights being the absolute abundance of the features in each density gradient.</p>
<p>Later, we will use the power of FDR corrections (using <a href="https://cran.r-project.org/web/packages/fdrtool/fdrtool.pdf">fdrtool</a>) and assess which were the incorporators.</p>
<p>The functions from <code>fdr_adjusted_pvalues.R</code> are used here. The input parameters required for this operation is the</p>
<ol style="list-style-type: decimal">
<li>Atom fraction excess table (<code>atomX</code>)</li>
<li>Significance value</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb5-1"></a>multiple_testing_table =<span class="st"> </span><span class="kw">fdr_multiple_hypothesis_testing</span>(<span class="dt">atomX =</span> atomX, <span class="dt">alpha =</span> <span class="fl">0.05</span>) </span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb6-1"></a>incorporator_list_MT =<span class="st"> </span>multiple_testing_table <span class="op">%&gt;%</span></span>
<span id="cb6-2"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb6-2"></a><span class="st"> </span><span class="kw">select</span>(OTU, incorporator) <span class="op">%&gt;%</span></span>
<span id="cb6-3"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb6-3"></a><span class="st">  </span><span class="kw">filter</span>(incorporator <span class="op">==</span><span class="st"> </span><span class="ot">TRUE</span>)</span>
<span id="cb6-4"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb6-4"></a></span>
<span id="cb6-5"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb6-5"></a>n_incorp_MT =<span class="st"> </span><span class="kw">nrow</span>(incorporator_list_MT)</span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb7-1"></a><span class="kw">cat</span>(<span class="st">&#39;Number of incorporators:&#39;</span>, n_incorp_MT, <span class="st">&#39;</span><span class="ch">\n</span><span class="st">&#39;</span>)</span></code></pre></div>
<pre><code>## Number of incorporators: 38</code></pre>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb9-1"></a>rmarkdown<span class="op">::</span><span class="kw">paged_table</span>(<span class="dt">x =</span> incorporator_list_MT)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["OTU"],"name":[1],"type":["chr"],"align":["left"]},{"label":["incorporator"],"name":[2],"type":["lgl"],"align":["right"]}],"data":[{"1":"NC_000913.3","2":"TRUE"},{"1":"NC_002737.2","2":"TRUE"},{"1":"NC_003450.3","2":"TRUE"},{"1":"NC_008261.1","2":"TRUE"},{"1":"NC_009012.1","2":"TRUE"},{"1":"NC_010067.1","2":"TRUE"},{"1":"NC_012982.1","2":"TRUE"},{"1":"NC_012983.1","2":"TRUE"},{"1":"NC_014008.1","2":"TRUE"},{"1":"NC_014168.1","2":"TRUE"},{"1":"NC_014210.1","2":"TRUE"},{"1":"NC_014211.1","2":"TRUE"},{"1":"NC_014212.1","2":"TRUE"},{"1":"NC_014213.1","2":"TRUE"},{"1":"NC_014214.1","2":"TRUE"},{"1":"NC_014363.1","2":"TRUE"},{"1":"NC_014364.1","2":"TRUE"},{"1":"NC_015761.1","2":"TRUE"},{"1":"NC_017033.1","2":"TRUE"},{"1":"NC_017095.1","2":"TRUE"},{"1":"NC_018014.1","2":"TRUE"},{"1":"NC_018066.1","2":"TRUE"},{"1":"NC_018067.1","2":"TRUE"},{"1":"NC_018068.1","2":"TRUE"},{"1":"NC_018515.1","2":"TRUE"},{"1":"NC_019792.1","2":"TRUE"},{"1":"NC_019897.1","2":"TRUE"},{"1":"NC_019898.1","2":"TRUE"},{"1":"NC_019904.1","2":"TRUE"},{"1":"NC_019936.1","2":"TRUE"},{"1":"NC_019937.1","2":"TRUE"},{"1":"NC_019938.1","2":"TRUE"},{"1":"NC_019939.1","2":"TRUE"},{"1":"NC_019964.1","2":"TRUE"},{"1":"NC_019974.1","2":"TRUE"},{"1":"NC_019975.1","2":"TRUE"},{"1":"NC_019976.1","2":"TRUE"},{"1":"NC_021184.1","2":"TRUE"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
<div id="comparison-of-the-two-methods-1" class="section level3">
<h3><span class="header-section-number">3.2.3</span> Comparison of the two methods</h3>
<p>In the simulated study, all genomes were set to be isotope incorporators. However, the AFE method does not identify all the incorporators simulated in the study. On the other hand, the method with multiple hypothesis testing could identify all incorporators.</p>
<p>While these tests cannot tell which method is superior, we would leave it to the user’s judgment to take a call on which method to use.</p>
<p>Additionally, one could combine the two methods to use the power of multiple testing along with the having the estimated isotopic enrichment.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb10-1"></a>AFE_incorporators_MT =<span class="st"> </span>multiple_testing_table <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">inner_join</span>(df_atomX_boot <span class="op">%&gt;%</span><span class="st"> </span><span class="kw">select</span>(<span class="op">-</span>Incorporator)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb10-2"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb10-2"></a><span class="st">  </span><span class="kw">select</span>(OTU, incorporator, A, A_CI_low, A_CI_high)</span>
<span id="cb10-3"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb10-3"></a></span>
<span id="cb10-4"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb10-4"></a>rmarkdown<span class="op">::</span><span class="kw">paged_table</span>(AFE_incorporators_MT)</span></code></pre></div>
<div data-pagedtable="false">
<script data-pagedtable-source type="application/json">
{"columns":[{"label":["OTU"],"name":[1],"type":["chr"],"align":["left"]},{"label":["incorporator"],"name":[2],"type":["lgl"],"align":["right"]},{"label":["A"],"name":[3],"type":["dbl"],"align":["right"]},{"label":["A_CI_low"],"name":[4],"type":["dbl"],"align":["right"]},{"label":["A_CI_high"],"name":[5],"type":["dbl"],"align":["right"]}],"data":[{"1":"NC_000913.3","2":"TRUE","3":"0.01736205","4":"0.01552447","5":"0.02050354"},{"1":"NC_002737.2","2":"TRUE","3":"-0.19158486","4":"-0.20538603","5":"-0.19259449"},{"1":"NC_003450.3","2":"TRUE","3":"0.05789565","4":"0.04490071","5":"0.07070187"},{"1":"NC_008261.1","2":"TRUE","3":"0.41607862","4":"0.41766195","5":"0.42578091"},{"1":"NC_009012.1","2":"TRUE","3":"-0.17073384","4":"-0.18287346","5":"-0.17182058"},{"1":"NC_010067.1","2":"TRUE","3":"0.02247530","4":"0.02130294","5":"0.02603644"},{"1":"NC_012982.1","2":"TRUE","3":"-0.06834989","4":"-0.07374299","5":"-0.06799555"},{"1":"NC_012983.1","2":"TRUE","3":"-0.06974998","4":"-0.07694446","5":"-0.06804840"},{"1":"NC_014008.1","2":"TRUE","3":"0.05956431","4":"0.05684992","5":"0.06539130"},{"1":"NC_014168.1","2":"TRUE","3":"0.19554493","4":"0.19759030","5":"0.20401995"},{"1":"NC_014210.1","2":"TRUE","3":"0.26642966","4":"0.27081270","5":"0.27476463"},{"1":"NC_014211.1","2":"TRUE","3":"0.25786593","4":"0.25802147","5":"0.27100233"},{"1":"NC_014212.1","2":"TRUE","3":"0.18272276","4":"0.17687558","5":"0.19699713"},{"1":"NC_014213.1","2":"TRUE","3":"0.19471633","4":"0.18474045","5":"0.22015860"},{"1":"NC_014214.1","2":"TRUE","3":"0.18594146","4":"0.17337145","5":"0.21362617"},{"1":"NC_014363.1","2":"TRUE","3":"0.18927007","4":"0.18412222","5":"0.20576905"},{"1":"NC_014364.1","2":"TRUE","3":"-0.03212923","4":"-0.03784406","5":"-0.02818252"},{"1":"NC_015761.1","2":"TRUE","3":"0.02504089","4":"0.02038013","5":"0.03025883"},{"1":"NC_017033.1","2":"TRUE","3":"0.18435708","4":"0.18104733","5":"0.19788006"},{"1":"NC_017095.1","2":"TRUE","3":"-0.11913907","4":"-0.13602138","5":"-0.11294674"},{"1":"NC_018014.1","2":"TRUE","3":"0.12586753","4":"0.12327090","5":"0.13700847"},{"1":"NC_018066.1","2":"TRUE","3":"-0.05180269","4":"-0.08690786","5":"-0.01268549"},{"1":"NC_018067.1","2":"TRUE","3":"-0.23378418","4":"-0.35059635","5":"-0.12862915"},{"1":"NC_018068.1","2":"TRUE","3":"-0.07864267","4":"-0.09099816","5":"-0.07204643"},{"1":"NC_018515.1","2":"TRUE","3":"-0.08186764","4":"-0.09299843","5":"-0.07871894"},{"1":"NC_019792.1","2":"TRUE","3":"0.15955810","4":"0.16024609","5":"0.16799815"},{"1":"NC_019897.1","2":"TRUE","3":"0.13114911","4":"0.13102388","5":"0.13941350"},{"1":"NC_019898.1","2":"TRUE","3":"-0.02850044","4":"-0.03832904","5":"-0.02336664"},{"1":"NC_019904.1","2":"TRUE","3":"-0.08011310","4":"-0.08622916","5":"-0.08007572"},{"1":"NC_019936.1","2":"TRUE","3":"0.16335602","4":"0.16418855","5":"0.17263780"},{"1":"NC_019937.1","2":"TRUE","3":"0.06355637","4":"0.05425472","5":"0.07260052"},{"1":"NC_019938.1","2":"TRUE","3":"0.10500672","4":"0.09985510","5":"0.11756926"},{"1":"NC_019939.1","2":"TRUE","3":"0.16826409","4":"0.15529904","5":"0.19082428"},{"1":"NC_019964.1","2":"TRUE","3":"0.18740820","4":"0.18807679","5":"0.19785145"},{"1":"NC_019974.1","2":"TRUE","3":"0.18153967","4":"0.18106033","5":"0.19149511"},{"1":"NC_019975.1","2":"TRUE","3":"0.07156263","4":"0.05167593","5":"0.09484591"},{"1":"NC_019976.1","2":"TRUE","3":"0.13371145","4":"0.13257695","5":"0.14283593"},{"1":"NC_021184.1","2":"TRUE","3":"-0.05132632","4":"-0.06436726","5":"-0.04309271"}],"options":{"columns":{"min":{},"max":[10]},"rows":{"min":[10],"max":[10]},"pages":{}}}
  </script>
</div>
</div>
</div>
<div id="session-information-2" class="section level2">
<h2><span class="header-section-number">3.3</span> Session information</h2>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="qsip-metagenomics-and-identification-of-isotope-incorporators.html#cb11-1"></a><span class="kw">sessionInfo</span>()</span></code></pre></div>
<pre><code>## R version 4.0.3 (2020-10-10)
## Platform: x86_64-w64-mingw32/x64 (64-bit)
## Running under: Windows 10 x64 (build 19042)
## 
## Matrix products: default
## 
## locale:
## [1] LC_COLLATE=English_Canada.1252  LC_CTYPE=English_Canada.1252   
## [3] LC_MONETARY=English_Canada.1252 LC_NUMERIC=C                   
## [5] LC_TIME=English_Canada.1252    
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] qSIPmg_0.3.0      ggpubr_0.4.0      data.table_1.14.2 HTSSIP_1.4.1     
##  [5] phyloseq_1.34.0   forcats_0.5.1     stringr_1.4.0     dplyr_1.0.7      
##  [9] purrr_0.3.4       readr_2.0.2       tidyr_1.1.4       tibble_3.1.5     
## [13] ggplot2_3.3.5     tidyverse_1.3.1  
## 
## loaded via a namespace (and not attached):
##   [1] colorspace_2.0-2    ggsignif_0.6.3      ellipsis_0.3.2     
##   [4] rio_0.5.27          XVector_0.30.0      fs_1.5.0           
##   [7] rstudioapi_0.13     farver_2.1.0        bit64_4.0.5        
##  [10] fansi_0.5.0         lubridate_1.8.0     xml2_1.3.2         
##  [13] codetools_0.2-18    splines_4.0.3       knitr_1.36         
##  [16] ade4_1.7-18         polynom_1.4-0       jsonlite_1.7.2     
##  [19] broom_0.7.9         cluster_2.1.2       dbplyr_2.1.1       
##  [22] compiler_4.0.3      httr_1.4.2          backports_1.2.1    
##  [25] lazyeval_0.2.2      assertthat_0.2.1    Matrix_1.3-4       
##  [28] fastmap_1.1.0       cli_3.0.1           htmltools_0.5.2    
##  [31] tools_4.0.3         igraph_1.2.7        gtable_0.3.0       
##  [34] glue_1.4.2          reshape2_1.4.4      Rcpp_1.0.5         
##  [37] carData_3.0-4       Biobase_2.50.0      cellranger_1.1.0   
##  [40] jquerylib_0.1.4     vctrs_0.3.8         Biostrings_2.58.0  
##  [43] rhdf5filters_1.2.0  multtest_2.46.0     ape_5.5            
##  [46] nlme_3.1-153        iterators_1.0.13    xfun_0.27          
##  [49] openxlsx_4.2.4      rvest_1.0.2         lifecycle_1.0.1    
##  [52] rstatix_0.7.0       zlibbioc_1.36.0     MASS_7.3-53        
##  [55] scales_1.1.1        vroom_1.5.5         hms_1.1.1          
##  [58] parallel_4.0.3      biomformat_1.18.0   rhdf5_2.34.0       
##  [61] yaml_2.2.1          curl_4.3.2          sass_0.4.0         
##  [64] stringi_1.5.3       S4Vectors_0.28.0    foreach_1.5.1      
##  [67] permute_0.9-5       BiocGenerics_0.36.1 zip_2.2.0          
##  [70] rlang_0.4.10        pkgconfig_2.0.3     evaluate_0.14      
##  [73] lattice_0.20-45     Rhdf5lib_1.12.0     labeling_0.4.2     
##  [76] bit_4.0.4           tidyselect_1.1.1    plyr_1.8.6         
##  [79] magrittr_2.0.1      bookdown_0.24       R6_2.5.1           
##  [82] IRanges_2.24.0      generics_0.1.1      DBI_1.1.1          
##  [85] pillar_1.6.4        haven_2.4.3         foreign_0.8-81     
##  [88] withr_2.4.2         mgcv_1.8-38         survival_3.2-13    
##  [91] abind_1.4-5         modelr_0.1.8        crayon_1.4.1       
##  [94] car_3.0-11          fdrtool_1.2.16      utf8_1.2.2         
##  [97] tzdb_0.1.2          rmarkdown_2.11      grid_4.0.3         
## [100] readxl_1.3.1        vegan_2.5-7         reprex_2.0.1       
## [103] digest_0.6.27       stats4_4.0.3        munsell_0.5.0      
## [106] bslib_0.3.1</code></pre>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="scaling-coverage-data.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "section"
},
"link": "GitHub repo link"
});
});
</script>

</body>

</html>
