```{r setup 3, include=FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
```
# qSIP metagenomics and identification of isotope incorporators

The isotope incorporators could be identified using two methods:

1.	Atom fraction excess (AFE) method. E.g. the Hungate et al. (2015) AFE estimation method.
2.	Multiple hypothesis testing to assess statistical significance 

While the first method gives a quantitative estimate of isotopic enrichment in a taxon to determine if it is an incorporator, the second method relies on null hypothesis testing of statistical significance between unlabeled and labeled treatments for a taxon.

In this vignette, we will compare the two methods using the data from the introductory vignette to qSIPmg. 

Please refer this study, for the AFE method proposed by [Hungate et al. (2015)]( https://journals.asm.org/doi/10.1128/AEM.02280-15)

For significance testing, we will use multiple hypothesis testing with Benjamini-Hohberg method to control the false discovery rate (FDR). FDR correction is performed based on the R package [fdrtool]( https://cran.r-project.org/web/packages/fdrtool/fdrtool.pdf). 

## Why perform multiple hypothesis testing with FDR correction?

For a detailed understanding of the issues with multiple testing, please refer the following works and more

1.	https://www.stat.berkeley.edu/~mgoldman/Section0402.pdf 

2.	https://genomebiology.biomedcentral.com/articles/10.1186/s13059-019-1716-1

3.	 https://mb3is.megx.net/gustame/warnings/warning-multiple-testing

Briefly, when the same statistical test is used to test several hypotheses simultaneously, the chance of falsely rejecting the null hypothesis (false positive rate or the rate of type 1 error) increases. This can be controlled using several methods as discussed in the above links. Of these methods, FDR correction is less conservative in controlling the probability of false positives, and thus has greater statistical power than other familywise error rate control methods.

Ok, now letâ€™s compare the two methods in identifying isotope incorporators

## Comparison of the two methods

### AFE method

Isotope incorporators from the introductory vignette were identified using the AFE method
Reproducing the code here, we have

```{r Load libraries 3, echo = F}
#Load required libraries
library(tidyverse)
library(phyloseq)
library(HTSSIP)
library(data.table)
library(ggpubr)
library(qSIPmg)
set.seed(seed = 1000)
```

```{r Load data 3, echo = F}
## Load data
#Coverage metadata
#Uncomment if your coverage data is in the format mentioned above for this file. Remains commented if you are using the output from `checkm coverage`
f_tibble <- read_csv("mock_input_data/coverage_metadata.csv")

#Sequins metadata
sequins <- read_csv(file="mock_input_data/sequins_metadata.csv")

#Dilutions data
seq_dil = read_csv(file = "mock_input_data/dilutions_data.csv")

#Log scale BOOLEAN. True or False depending on how you would want the MAG coverages to be scaled. Select TRUE if you need MAG concentrations scaled on the log scale
log_scale = TRUE

#coe_of_variation. Acceptable coefficient of variation for coverage and detection (eg. 20 - for 20 % threshold of coefficient of variation) (Coverages above the threshold value will be flagged in the plots)
coe_of_variation = 20 

#Taxonomy
gtdbtk_bac_summary = read_delim("mock_input_data/gtdbtk.bac120.summary.tsv", 
                                 "\t", escape_double = FALSE, trim_ws = TRUE)
gtdbtk_archaea = read_delim("mock_input_data/gtdbtk.ar122.summary.tsv", 
                             "\t", escape_double = FALSE, trim_ws = TRUE)
#GC content
GC_content <- read_csv(file = "mock_input_data/GC_content.csv")

#Fractions
fractions = read_csv("mock_input_data/fractions.csv")

```

```{r estimate absolute concentrations 2, echo = F}
taxonomy_tibble = rbind(gtdbtk_bac_summary, gtdbtk_archaea) #Combine bacteria and archaea taxonomy files if it has not been done yet
#mag_tab is a tibble with absolute concentrations of MAGs obtained by scaling MAG coverages using linear regression models on sequin coverages and concentration

##Scale MAG coverages to obtain MAG absolute concentrations and save scaling plots in the working directory
#For rlm scaling using scale_features_rlm
#For rlm scaling using scale_features_lm
mag_tab_scaled <- scale_features_rlm(f_tibble, sequins, seq_dil, log_scale, coe_of_variation = coe_of_variation)
mag_tab = as.matrix(mag_tab_scaled$mag_tab) #Extract absolute abundances as a matrix
```

```{r Make phyloseq objects 2, echo = F}

mag.table = phyloseq::otu_table(mag_tab, taxa_are_rows = TRUE) #Phyloseq OTU table

taxonomy.object = tax.table(taxonomy_tibble) # Create a taxonomy phyloseq object
samples.object = sample.table(fractions) # Create a samples phyloseq object
phylo.qSIP = phylo.table(mag.table, taxonomy.object, samples.object) # Make a phyloseq table for downstream qSIP analysis
```

```{r AFE methodGet bootstrapped AFE table 2}
atomX = qSIP_atom_excess_MAGs(phylo.qSIP,
                               control_expr='Isotope=="12C"',
                               treatment_rep='Replicate',
                               Gi = GC_content)
#Bootstrap confidence intervals
df_atomX_boot = qSIP_bootstrap(atomX, n_boot=100, parallel = TRUE) #Change "parallel = FALSE" to compute using a single-core
CI_threshold = 0
df_atomX_boot = df_atomX_boot %>%
  mutate(Incorporator = A_CI_low > CI_threshold,
         OTU = reorder(OTU, -A))
```

List out the incorporators

```{r list incorporators 2}
#Get incorporator info
n_incorp_AFE = df_atomX_boot %>%
  filter(Incorporator == TRUE) %>%
  nrow 
#Get incorporator list
incorporator_list_AFE = incorporators_taxonomy(taxonomy = taxonomy_tibble, bootstrapped_AFE_table = df_atomX_boot)
#Print incorporator information
cat('Number of incorporators:', n_incorp_AFE, '\n')
rmarkdown::paged_table(incorporator_list_AFE)
```


### Multiple testing

The atom fraction excess table obtained in the introductory vignette has the centroid buoyant densities as well. In this method, we will perform t-tests on the centroid buoyant densities. Centroid buoyant densities are weighted mean buoyant density for a taxon, the weights being the absolute abundance of the features in each density gradient. 

Later, we will use the power of FDR corrections (using [fdrtool]( https://cran.r-project.org/web/packages/fdrtool/fdrtool.pdf)) and assess which were the incorporators.

The functions from `fdr_adjusted_pvalues.R` are used here. The input parameters required for this operation is the 

1.	Atom fraction excess table (`atomX`)
2. Significance value


``` {r FDR multiple hypothesis testing, comment=FALSE, message=FALSE, warning=FALSE, fig.show = "hide"}

multiple_testing_table = fdr_multiple_hypothesis_testing(atomX = atomX, alpha = 0.05) 

incorporator_list_MT = multiple_testing_table %>%
 select(OTU, incorporator) %>%
  filter(incorporator == TRUE)

n_incorp_MT = nrow(incorporator_list_MT)

```

```{r}
cat('Number of incorporators:', n_incorp_MT, '\n')

rmarkdown::paged_table(x = incorporator_list_MT)
```

### Comparison of the two methods

In the simulated study, all genomes were set to be isotope incorporators. However, the AFE method does not identify all the incorporators simulated in the study. On the other hand, the method with multiple hypothesis testing could identify all incorporators. 

While these tests cannot tell which method is superior, we would leave it to the user's judgment to take a call on which method to use.

Additionally, one could combine the two methods to use the power of multiple testing along with the having the estimated isotopic enrichment.

```{r combining the two methods}
AFE_incorporators_MT = multiple_testing_table %>% inner_join(df_atomX_boot %>% select(-Incorporator)) %>% 
  select(OTU, incorporator, A, A_CI_low, A_CI_high)

rmarkdown::paged_table(AFE_incorporators_MT)

```

## Session information

```{r}
sessionInfo()
```
