---
title: "qSIP_analysis"
author: "Pranav_Sampara"
date: "02/07/2020"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE)
```

# Introduction
qSIP analysis using sequins, MAG coverage, and MAG taxonomy is performed using this R markdown. Briefly, coverages are normalized, MAG coverages are then scaled based on linear regression models from sequin coverage and concentrations, a phyloseq object is created from MAG absolute concentrations and taxonomy (GTDB taxonomy output format is required) data.
Note: Sequins that were spiked in the DNA-SIP fractions will be used in scaling and creating linear regression models for evaluating absolute MAG concentrations. Please see `sequin_scaling.R`
For normalizing coverage values please see `pooling_funs.R`

This R markdown primarily uses Tidyverse, phyoseq, and HTSSIP packages
This markdown uses functions from functions_qSIP_MAGs.R
```{r Load libraries}
#Load required libraries
library(tidyverse)
library(phyloseq)
library(HTSSIP)
library(readxl)
library(data.table)
library(ggpubr)
```

## Load required data 

The following files are required:

#MAG coverage data
-Pooled coverages data across samples for `Features` incuding sequins that were used as spike-ins. The followings columns are required
    - Feature: A character string describing the `Feature` label
    - Sample: The label for these n number of columns should be in the format of "'isotope'_rep_#_fraction_#". For instance, "12C_rep_1_fraction_1". The number of sample columns should be the same as the product of (replicates, fractions, isotopes)
    
#Sequin metadata
-Load the sequins metadata which has the following columns
    -Feature: As described above
    -Length: Length of the sequin in bp
    -GC_content: GC content of the sequin
    -Sequence: Sequin nucleotide sequence
    -Concentration: Concentration ofhte sequin in attamoles/uL
    
#Dilutions data
-Load dilutions data that contains the following columns
    -Sample: Similar to the sample name as described above
    -Dilution: Dilution of sequins added to the fraction before sequencing.

#Fractions metadata
-A fractions file with the following columns
    - Replicate: Depends on how many replicates the study has
    - Fractions: Typically in the range of 1-24
    - Buoyant_density: As calculated from the refractometer for each fraction and replicate
    - Isotope - "12C", "13C", "14N", "15N" etc.
    - DNA_concentration
    - Sample - In the format "'isotope'_rep_#_fraction_#". For instance 12C_rep_1_fraction_1

#GTDB style taxonomy data
- A taxonomy file in the GTDB output format. Load the bacteria and archaea taxonomy outputs separately. The markdown requires loading the standard output files from GTDB-Tk separately for bacteria and archaea

#MAG absolute concentrations
- MAG absolute concentrations obtained from scaling_sequins.R. 
`mag_tab` object obtained from the above script is to be used as the input here

#GC content
- GC content of the MAGs. The table should contain the following columns
    - MAG: MAG identifier such as the `Feature` label from the sequin_scaling.R script
    - GC_content: GC content of the `Feature` in the range of 0-1
```{r Load data}
## Load data
#Coverage metadata
f_tibble = read_csv(file="coverage_metadata.csv")

#Sequins metadata
sequins <- read_csv(file="sequins_metadata.csv")

#Dilutions data
seq_dil = read_csv(file = "dilutions_data.csv")

#Log scale BOOLEAN. True or False depending on how you would want the MAG coverages to be scaled. Select TRUE if you need MAG concentrations scaled on the log scale
log_scale = TRUE

#Taxonomy
gtdbtk_bac_summary = read_delim("gtdbtk.bac120.summary.tsv", 
                                 "\t", escape_double = FALSE, trim_ws = TRUE)
gtdbtk_archaea = read_delim("gtdbtk_archaea.tsv", 
                             "\t", escape_double = FALSE, trim_ws = TRUE)
#GC content
GC_content <- read_delim("GC_content.tsv", 
    "\t", escape_double = FALSE, trim_ws = TRUE)

#Fractions
fractions_ps = read_csv("fractions_ps.csv")
```

## Source required functions

Source the "sequin_scaling.R" script. Please see source file ("sequin_scaling.R") for more details on scaling MAG coverages based on linear regression models on sequin coverages and concentrations, obtaining plots, and limit of detection
Create a local environment to load qSIP analysis functions. The local environment helps in moving between edited HTSSIP functions and the original HTSSIP package. Please see source file ("functions_qSIP_MAGs.R") for more information on the edited functions

```{r Source functions}
env = new.env()
environment()
source("sequin_scaling.R")
source("functions_qSIP_MAGs.R", local = env)
```

#qSIP analysis

Create required phyloseq objects 

```{r Make phyloseq objects}
taxonomy_ps = rbind(gtdbtk_bac_summary, gtdbtk_archaea) #Combine bacteria and archaea taxonomy files if necessary
#mag_tab is a tibble with absolute concentrations of MAGs obtained by scaling MAG coverages using linear regression models on sequin coverages and concentration

#Scale MAG coverages to obtain MAG absolute concentrations
mag_tab_scaled <- scale_features(f_tibble, sequins, seq_dil, log_scale)

mag_tab = as.matrix(mag_tab_scaled$mag_tab) #Extract absolute abundances as a matrix
#mag_tab = as.matrix(mag_tab) #Ensuring input for phyloseq OTU table is in a matrix format
mag.table = phyloseq::otu_table(mag_tab, taxa_are_rows = TRUE) #Phyloseq OTU table

taxonomy.object = env$tax.table(taxonomy_ps) # Create a taxonomy phyloseq object
samples.object = env$sample.table(fractions_ps) # Create a samples phyloseq object
phylo.qSIP = env$phylo.table(mag.table, taxonomy.object, samples.object) # Make a phyloseq table for downstream qSIP analysis
```

Calculate atom fraction excess. By bootstrapping confidence intervals, determine incorporators

```{r Calculate atom fraction excess}
atomX = env$qSIP_atom_excess(phylo.qSIP,
                               control_expr='Isotope=="12C"',
                               treatment_rep='Replicate',
                               Gi = GC_content)
#Bootstrap confidence intervals
df_atomX_boot = qSIP_bootstrap(atomX, n_boot=100)
df_atomX_boot %>% head
CI_threshold = 0
df_atomX_boot = df_atomX_boot %>%
  mutate(Incorporator = A_CI_low > CI_threshold,
         OTU = reorder(OTU, -A))
#Get incorporator info
n_incorp = df_atomX_boot %>%
  filter(Incorporator == TRUE) %>%
  nrow 
cat('Number of incorporators:', n_incorp, '\n')
```

Plot the atom fraction excess plot

```{r Plot atom fraction excess}
(atom_f_excess_plot = ggplot(df_atomX_boot, aes(OTU, A, ymin=A_CI_low, ymax=A_CI_high, color=Incorporator)) +
  geom_pointrange(size=0.25) +
  geom_linerange() +
  geom_hline(yintercept=0, linetype='dashed', alpha=0.5) +
  labs(x='MAGs', y='Atom fraction excess') +
  theme_bw() +
  coord_flip() +
  ggtitle("Isotope incorporating MAGs"))
```
